lib/src/ffmpeg/ffmpeg_command.dart

```diff
@@ -159,20 +162,20 @@ class FilterGraph {
 /// those filters then produce some number of output streams.
 class FilterChain {
   const FilterChain({
-    required this.inputs,
+    this.inputs,
     required this.filters,
-    required this.outputs,
+    this.outputs,
   });
 
   /// Streams that flow into the [filters].
-  final List<FfmpegStream> inputs;
+  final List<FfmpegStream>? inputs;
```

Comment:
Why are these nullable?

Reply:
Not all filters require inputs, such as subtitle filter, crop filter, fade filter, afade filter (although in some cases they might) it remains a better solution to omit these variables instead of giving an empty list to get around this which I noticed I had to do before.

Reply:
>  it remains a better solution to omit these variables instead of giving an empty list to get around this which I noticed I had to do before

I don't follow. Why is `null` better than an empty list?

Reply:
When using the ffmpeg cli, you wouldn't input anything if you did not want it to be included but when using the past solution of this library you would have to input an empty value just to specify that you did not want anything to be included in the final command whereas when using ffmpeg on command line you would omit the value instead. <br>
It also keeps more closely with the fact that other attributes are nullable inside of each of the filters - just keeping the library more consistent throughout.

Reply:
It's easy to solve the "required" problem without making it nullable:

```dart
const FilterChain({
  this.inputs = const [],
});
```

I'm concerned that in this particular case, by making lists nullable, we're adding a responsibility to all uses of the inputs and outputs that could otherwise treat empty lists and non-empty lists the same way.

For example, if somebody wants to spread the values, that works the same way whether the list is empty or non-empty. If somebody wants to join the values into a single string, that works the same way whether the list is empty or non-empty. But if the list, itself, is nullable, then all users have to explicitly handle that possibility.

We can see an example of this in your update to the `toCli()`. Because you avoided empty lists for `inputs` and `output`s you're now declaring empty lists to run the `map()` behavior.